### PR TITLE
feature/faster-testing

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description
Improve the speed of testing
Before 24m55.612s
After 4m56

# Impacts
running tests

# NEW Package(s) required
Test::ParallelSubtest

# Delete branch after merge
 NO

### CODE REVIEW

File: lib/pf/log/trapper.pm [hunk 1]
```diff
@@ -40,6 +40,8 @@ sub PRINTF {
     $self->PRINT($buf);
 }
 
+sub FILENO { undef }
```
Comment (extrafu) [inline lib/pf/log/trapper.pm]: Document why you do that.

**Comment by jrouzierinverse** [comment]
Updated

File: lib/pf/log/trapper.pm [hunk 2]
```diff
@@ -28,18 +29,39 @@ sub TIEHANDLE {
...
+
 sub PRINTF {
     my $self = shift;
     my $buf = sprintf(@_);
     $self->PRINT($buf);
 }
 
+=head2 FILENO
+
+Return undef to avoid Cache::BDB failing somethings
```
Comment (julsemaan) [inline lib/pf/log/trapper.pm]: failing somethings ? what things ?

or did you mean sometimes ?

File: lib/pfconfig/backend/memory.pm [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+sub init {
+    my ($self) = @_;
+    $self->{cache} = CHI->new(driver => 'Memory', datastore => {},'serializer' => 'Sereal');
+}
+
+sub set {
+    my ( $self, $key, $value ) = @_;
+
+    # BDB doesn't write empty strings
```
Comment (julsemaan) [inline lib/pfconfig/backend/memory.pm]: that's not BDB though, that CHI isn't it ?

File: lib/pfconfig/backend/memory.pm [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    # We workaround it using a class that represents an empty string
+    if ( defined($value) && "$value" eq '' ) {
```
Comment (julsemaan) [inline lib/pfconfig/backend/memory.pm]: Is this really necessary for CHI

File: lib/pfconfig/backend/memory.pm [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    $self->SUPER::set( $key, $value );
+}
+
+sub get {
+    my ( $self, $key ) = @_;
+    my $value = $self->SUPER::get($key);
+
+    # BDB doesn't write empty strings
+    # We workaround it using a class that represents an empty string
+    if ( ref($value) eq "pfconfig::empty_string" && $value->isa("pfconfig::empty_string") ) {
```
Comment (julsemaan) [inline lib/pfconfig/backend/memory.pm]: Same about whether it is necessary for CHI

File: sbin/pfconfig [hunk 6]
```diff
@@ -411,6 +417,27 @@ sub normal_sighandler {
     $RUNNING = 0;
 }
 
+sub getLockOnPid {
```
Comment (julsemaan) [inline sbin/pfconfig]: perldoc

**Review by julsemaan — COMMENTED** [review]
Perl + golang tests are still passing with this PR, so other than the noted adjustments, this should be good to merge

File: lib/pfconfig/backend/memory.pm (hunk 4 above)
Comment (jrouzierinverse) [reply lib/pfconfig/backend/memory.pm]: Unfortunately yes

**Pull request merged**